src/analysis/retail/shaman/enhancement/modules/apl/StormbringerStorm.tsx

```diff
@@ -0,0 +1,87 @@
+import SPELLS from 'common/SPELLS/shaman';
+import TALENTS from 'common/TALENTS/shaman';
+import SpellLink from 'interface/SpellLink';
+import Combatant from 'parser/core/Combatant';
+import { Apl, build, Rule } from 'parser/shared/metrics/apl';
+import {
+  and,
+  buffPresent,
+  debuffMissing,
+  describe,
+  hasTalent,
+  repeatableBuffPresent,
+} from 'parser/shared/metrics/apl/conditions';
+import { AtLeastFiveMSW, MaxStacksMSW, minimumMaelstromWeaponStacks } from './Conditions';
+
+export function stormbringerStorm(combatant: Combatant): Apl {
+  const rules: (Rule | false)[] = [
+    {
+      spell: SPELLS.TEMPEST_CAST,
+      condition: describe(and(buffPresent(SPELLS.TEMPEST_BUFF), MaxStacksMSW), () => (
+        <>
+          available and at 10 <SpellLink spell={SPELLS.MAELSTROM_WEAPON_BUFF} /> stacks
+        </>
+      )),
+    },
+    {
+      spell: SPELLS.WINDSTRIKE_CAST,
+      condition: describe(
+        buffPresent(TALENTS.ASCENDANCE_ENHANCEMENT_TALENT),
+        () => (
+          <>
+            on cooldown during <SpellLink spell={TALENTS.ASCENDANCE_ENHANCEMENT_TALENT} />
+          </>
+        ),
+        '',
+      ),
+    },
+    {
+      spell: TALENTS.ELEMENTAL_BLAST_ELEMENTAL_TALENT,
+      condition: and(
+        minimumMaelstromWeaponStacks(8),
+        repeatableBuffPresent(
+          [
+            SPELLS.ELEMENTAL_SPIRITS_BUFF_MOLTEN_WEAPON,
+            SPELLS.ELEMENTAL_SPIRITS_BUFF_ICY_EDGE,
+            SPELLS.ELEMENTAL_SPIRITS_BUFF_CRACKLING_SURGE,
+          ],
+          { atLeast: 4 },
+        ),
+      ),
+    },
+    {
+      spell: SPELLS.LIGHTNING_BOLT,
+      condition: MaxStacksMSW,
+    },
+    {
+      spell: TALENTS.STORMSTRIKE_TALENT,
+      condition: describe(
+        hasTalent(TALENTS.DEEPLY_ROOTED_ELEMENTS_TALENT),
+        () => (
+          <>
+            to fish for <SpellLink spell={TALENTS.DEEPLY_ROOTED_ELEMENTS_TALENT} />
+          </>
+        ),
+        '',
+      ),
+    },
+    {
+      spell: SPELLS.LIGHTNING_BOLT,
+      condition: AtLeastFiveMSW,
+    },
+    TALENTS.ICE_STRIKE_TALENT,
+    {
+      spell: TALENTS.FROST_SHOCK_TALENT,
+      condition: buffPresent(SPELLS.HAILSTORM_BUFF),
+    },
+    TALENTS.SUNDERING_TALENT,
+    {
+      spell: SPELLS.FLAME_SHOCK,
+      condition: debuffMissing(SPELLS.FLAME_SHOCK),
+    },
+    TALENTS.LAVA_LASH_TALENT,
+    !combatant.hasTalent(TALENTS.HAILSTORM_TALENT) && TALENTS.FROST_SHOCK_TALENT,
```

Comment:
is this "the spell is literally not castable" or "the spell should never be cast"?

Reply:
This is a "only include the APL step if hailstorm isn't talented". If hailstorm _is_ picked, frost shock is cast higher up in the APL

Reply:
On second thoughts, I've fixed it